internal/transformations/remove_nulls.go

```diff
@@ -8,6 +8,7 @@ import (
 )
 
 // removeNulls removes NUL bytes in input.
-func removeNulls(data string) (string, error) {
-	return strings.ReplaceAll(data, "\x00", ""), nil
+func removeNulls(data string) (string, bool, error) {
+	transformedData := strings.ReplaceAll(data, "\x00", "")
+	return transformedData, data != transformedData, nil
```

Comment:
Can´t we do the comparison based in length? It is clear a successful replacement decreases the string length.